autofill/autofill-api/src/main/java/com/duckduckgo/autofill/api/AutofillCapabilityChecker.kt

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2023 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.autofill.api
+
+/**
+ * Used to check the status of various Autofill features.
+ *
+ * Whether autofill features are enabled depends on a variety of inputs. This class provides a single way to query the status of all of them.
+ */
+interface AutofillCapabilityChecker {
```

Comment:
I don't think any of these APIs should be `suspend`. Suspend is not a threading mechanism and we're wrongly using to "enforce" threading.
Suspend doesn't enforce threading, and now however we're letting `suspend` to pollute the callers.

Reply:
As discussed offline, there can be a larger conversation about APIs and `suspend`, but since `autofill` is already built around async, the current project is not the place to resolve that discussion or bring in wider changes to it.